src/azure/utils.ts

```diff
@@ -140,21 +140,25 @@ export async function fetchUserGroups(userId: string): Promise<UserGroup[]> {
     }
 
     let groups: UserGroup[] = [];
-    let nextUrl: string | undefined = uri.toString();
-    while (nextUrl) {
-        const response = await httpHelper.makeGetRequest<{
-            value: UserGroup[];
-            "@odata.nextLink"?: string;
-        }>(nextUrl, token);
-
-        const result = response.data.value.map(
-            (group) => ({ displayName: group.displayName, id: group.id }) as UserGroup,
-        );
+    try {
```

Comment:
If you move the try-catch to inside the `while` loop (but still containg `makeGetRequest`, then you'll get better resilience because it won't be an all-or-nothing operation to populate the list of groups.